src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3847,12 +3841,51 @@ protected Object multiBytes(DynamicObject string, int characterIndex,
 
             // TODO (nirvdrum 22-Dec-16): Since we specialize elsewhere on index being too large, do we need this? Can character boundary search in a CR_BROKEN string cause us to encounter this case?
             if (indexTooLargeProfile.profile(i < k)) {
-                return nil();
+                return -1;
             } else {
                 return p;
             }
         }
 
+        protected boolean characterIndexTooLarge(Rope rope, int characterIndex) {
+            return characterIndex > rope.characterLength();
+        }
+
+        protected boolean characterIndexInBounds(Rope rope, int characterIndex) {
+            return characterIndex >= 0 && !characterIndexTooLarge(rope, characterIndex);
```

Comment:
I would inline `characterIndexTooLarge` here since it's not used anywhere else.

Reply:
Will do.

Reply:
Not done, but it is very minor.